src/main/sensors/compass.c

```diff
@@ -54,35 +58,114 @@ mag_t mag;                   // mag access functions
 #define COMPASS_INTERRUPT_TAG   IO_TAG_NONE
 #endif
 
-PG_REGISTER_WITH_RESET_TEMPLATE(compassConfig_t, compassConfig, PG_COMPASS_CONFIG, 0);
+PG_REGISTER_WITH_RESET_FN(compassConfig_t, compassConfig, PG_COMPASS_CONFIG, 0);
 
-PG_RESET_TEMPLATE(compassConfig_t, compassConfig,
-    .mag_align = ALIGN_DEFAULT,
-    // xxx_hardware: 0:default/autodetect, 1: disable
-    .mag_hardware = 1,
-    .mag_declination = 0,
-    .interruptTag = COMPASS_INTERRUPT_TAG
-);
+void pgResetFn_compassConfig(compassConfig_t *compassConfig)
```

Comment:
It is possible to `#define` initialization values and then use them in `PG_RESET_TEMPLATE`

Reply:
Eliminated all static variables in the drivers.